tiramisu_programs/tiramisu_program.py

```diff
@@ -228,6 +227,14 @@ def call_solver(self, comp, params):
             to_replace = re.findall(r'(std::ofstream out(?s:.)+)return',
                                     original_str)[0]
             header = "function * fct = tiramisu::global::get_implicit_function();\n"
+
+            original_str = original_str.replace(
```

Comment:
Instead of writing legality and then delete it, why not just remove it from start? I believe the proper way to do things is to not generate legality in the first place.

Reply:
I tried to keep the changes to a minimum.

Doing what you described means starting from the original code and then adding the code to apply the transformations. Starting from the legality check code the transformations are already applied.

Computation wise they are equivalent. I decided to keep the original author's decision and just tweaked it to remove the legality checks.

Good insight though